tdrs-frontend/src/components/FileUpload/FileUpload.jsx

```diff
@@ -7,35 +7,63 @@ import {
   clearFile,
   SET_FILE_ERROR,
   upload,
+  download,
 } from '../../actions/reports'
+
+import Button from '../Button'
+
 import createFileInputErrorState from '../../utils/createFileInputErrorState'
+import { handlePreview, getTargetClassName } from './utils'
 
 const INVALID_FILE_ERROR =
   'We can’t process that file format. Please provide a plain text file.'
 
 function FileUpload({ section, setLocalAlertState }) {
   // e.g. 'Aggregate Case Data' => 'aggregate-case-data'
   // The set of uploaded files in our Redux state
-  const files = useSelector((state) => state.reports.files)
+  const { files, year, downloadedFile } = useSelector(
+    (state) => state.reports
+  )
+
   const dispatch = useDispatch()
 
   // e.g. "1 - Active Case Data" => ["1", "Active Case Data"]
   const [sectionNumber, sectionName] = section.split(' - ')
 
+  const hasFile = files?.some((file) => {
+    return file.section === sectionName && file.uuid
+  })
+
   const selectedFile = files.find((file) => sectionName === file.section)
 
   const formattedSectionName = selectedFile.section
     .split(' ')
     .map((word) => word.toLowerCase())
     .join('-')
 
-  const fileName = selectedFile?.fileName
+  const targetClassName = getTargetClassName(formattedSectionName)
+
+  const fileName = selectedFile?.fileName || 'report.txt'
   const hasUploadedFile = Boolean(fileName)
 
   const ariaDescription = hasUploadedFile
     ? `Selected File ${selectedFile?.fileName}. To change the selected file, click this button.`
     : `Drag file here or choose from folder.`
 
+  useEffect(() => {
+    const trySettingPreview = () => {
+      const previewState = handlePreview(fileName, targetClassName)
+      if (!previewState) {
+        setTimeout(trySettingPreview, 100)
```

Comment:
This is something else I was kind of uncomfortable doing, but we don't have a lot of other options given that uswds is doing state updates outside of the typical react rendering flow, so we have to handle this bit of rendering as an effect, and we don't actually have a real idea of when this is supose to happen, because the state update/render is happening inside of their library in a private manner

Reply:
This is appropriate given the constraints.